src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -793,8 +799,83 @@ internal List<INodeLibraryDependencyInfo> NodeLocalDefinitions
             }
         }
 
+        internal List<INodeLibraryDependencyInfo> ExternalFiles
+        {
+            get
+            {
+                return GetExternalFiles();
+            }
+            set
+            {
+                foreach (var dependency in value)
+                {
+                    if (dependency.ReferenceType == ReferenceType.External)
+                    {
+                        foreach (var node in dependency.Nodes)
+                        {
+                            externalFilesDictionary[node] = dependency as DependencyInfo;
+                        }
+                    }
+                }
+
+                RaisePropertyChanged(nameof(ExternalFiles));
+            }
+        }
+
+        private List<INodeLibraryDependencyInfo> GetExternalFiles()
+        {
+            var externalFiles = new Dictionary<object, DependencyInfo>();
+
+            // Computes the external file references if the Workspace Model is a HomeWorkspaceModel.
+            // The workspace should be executed for the external references to be computed because the node output port values are needed.
+            if (this is HomeWorkspaceModel homeWorkspaceModel)
```

Comment:
Can you work with @Jingyi-Wen for a way to show indication about this limitation? Unless it is already implemented

Reply:
Yes, will look into it. Also @Jingyi-Wen Do we have any icons for the external file references?